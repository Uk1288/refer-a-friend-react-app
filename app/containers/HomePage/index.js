import React from 'react';
import styled from 'styled-components';
import Header from '../../components/header/index';
import SubHeader from '../../components/subheader/index';
import Summary from '../../components/summary/index';
import ReferAFriendForm from '../forms/refer-a-friend/index';
import LogoLoader from '../../components/logo-loader/index';

const ParentDiv = styled.div`
  margin: 16px;
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 260px;
`;

export default function HomePage() {
  return (
    <ParentDiv>
      <Header />
      <SubHeader />
      <Summary />
      <ReferAFriendForm />
      <LogoLoader />
    </ParentDiv>
  );
}
