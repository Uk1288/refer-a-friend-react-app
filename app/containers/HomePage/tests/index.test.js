import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';
import HomePage from '../index';
import { rootStateMock } from '../../../mocks/root.mock';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore(rootStateMock, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
