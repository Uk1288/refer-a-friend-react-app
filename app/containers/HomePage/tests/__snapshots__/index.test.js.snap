// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="HomePage__ParentDiv-sc-1dwbuoi-0 jaZlZw"
>
  <h1
    class="header__StyledHeader-sc-1mu6x1k-0 griTSl"
  >
    <span>
      Refer a friend
    </span>
  </h1>
  <h1
    class="subheader__StyledSubHeader-sc-1ur12bv-0 fByRsJ"
  >
    <span>
      Invite friends, get discount code
    </span>
  </h1>
  <h1
    class="summary__StyledSummary-sc-1spj6xq-0 hOYGGb"
  >
    <span>
      Refer your friends to ClassRT for you to both receive a 20% discount on your
    next purchased class. Your code will be sent to you once your referrals complete 
    their first purchase. Happy learning!
    </span>
  </h1>
  <div>
    <form>
      <div
        class="refer-a-friend__StyledEmailText-sc-16r8705-0 hDQkIh"
      >
        <span>
          Enter email address
        </span>
      </div>
      <div
        class="refer-a-friend__StyledInputBox-sc-16r8705-1 Qzlag"
      >
        <input
          class="input__StyledInput-sc-10zxn30-0 glFmJb"
          id="email"
          name="email"
          value=""
        />
        <button
          class="button__StyledButton-sc-1bg6su4-0 hygGZF"
          type="submit"
        >
          <span>
            Send
          </span>
        </button>
      </div>
    </form>
  </div>
  <div>
    <div
      class="logo-loader__StyledImageParent-sc-1r2a1aj-1 brbqgv"
    >
      <img
        alt="Refer a friend logo"
        class="logo-loader__StyledImage-sc-1r2a1aj-0 dZGvMc"
        src="IMAGE_MOCK"
      />
    </div>
  </div>
</div>
`;
